src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Frameworks/SupportedTargetFrameworkAliasProvider.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+using Microsoft.Build.Framework.XamlTypes;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.Threading;
+using EnumCollection = System.Collections.Generic.ICollection<Microsoft.VisualStudio.ProjectSystem.Properties.IEnumValue>;
+using EnumCollectionProjectValue = Microsoft.VisualStudio.ProjectSystem.IProjectVersionedValue<System.Collections.Generic.ICollection<Microsoft.VisualStudio.ProjectSystem.Properties.IEnumValue>>;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Frameworks
+{
+    /// <summary>
+    ///     Responsible for producing valid values for the TargetFramework property from a design-time build.
+    /// </summary>
+    [ExportDynamicEnumValuesProvider("SupportedTargetFrameworkAliasEnumProvider")]
+    [AppliesTo(ProjectCapability.DotNet)]
+    internal class SupportedTargetFrameworkAliasProvider : ChainedProjectValueDataSourceBase<EnumCollection>, IDynamicEnumValuesProvider, IDynamicEnumValuesGenerator
```

Comment:
Might be good to give the class the same name as the exported contract, to make it easier to find (`SupportedTargetFrameworkAliasEnumProvider`).

Reply:
Good suggestion, I'll make a note to do the same for the other TargetFramework-related EnumProviders.